### PR TITLE
Use realistic examples in User Guide

### PR DESCRIPTION
**Pull request by psychsane**
This pull request is created to update the examples in the official documentation.

Issue: #1651

## Overview

- Added sample Calculator class

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](http://junit.org/junit5/docs/snapshot/api/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [x] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [x] Change is documented in the [User Guide](http://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](http://junit.org/junit5/docs/snapshot/user-guide/#release-notes)
- [x] All [continuous integration builds](https://github.com/junit-team/junit5#continuous-integration-builds) pass

### CODE REVIEW

File: documentation/src/test/java/example/ParameterizedTestDemo.java [hunk 1]
```diff
@@ -10,11 +10,7 @@
 
 package example;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertFalse;
-import static org.junit.jupiter.api.Assertions.assertNotEquals;
-import static org.junit.jupiter.api.Assertions.assertNotNull;
-import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.*;
```
[the remark this answers is not in the document]
Comment (psychsane) [reply documentation/src/test/java/example/ParameterizedTestDemo.java]: Sorry, fixing it!

File: documentation/src/test/java/example/util/Calculator.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package example.util;
+
+public class Calculator {
+
+	public static int add(int a, int b) {
```
[the remark this answers is not in the document]
Comment (psychsane) [reply documentation/src/test/java/example/util/Calculator.java]: Okie Dokie!

File: documentation/src/test/java/example/ParameterizedTestDemo.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (psychsane) [reply documentation/src/test/java/example/ParameterizedTestDemo.java]: Fixed!

File: documentation/src/test/java/example/util/Calculator.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (psychsane) [reply documentation/src/test/java/example/util/Calculator.java]: Done!

File: documentation/src/test/java/example/HamcrestAssertionDemo.java [hunk 3]
```diff
@@ -11,17 +11,22 @@
...
 import org.junit.jupiter.api.Test;
 
 class HamcrestAssertionDemo {
 
+	private final Calculator calculator = new Calculator();
+
 	@Test
 	void assertWithHamcrestMatcher() {
-		assertThat(2 + 1, is(equalTo(3)));
+		assertThat(2 + 1, is(equalTo(calculator.subtract(4, 1))));
```
Comment (sbrannen) [inline documentation/src/test/java/example/HamcrestAssertionDemo.java]: Please replace `2 + 1` with `3`.

File: documentation/src/test/java/example/ParameterizedTestDemo.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (sbrannen) [reply documentation/src/test/java/example/ParameterizedTestDemo.java]: Thanks

File: documentation/src/test/java/example/util/Calculator.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (sbrannen) [reply documentation/src/test/java/example/util/Calculator.java]: Thanks

File: documentation/src/test/java/example/HamcrestAssertionDemo.java (hunk 3 above)
Comment (psychsane) [reply documentation/src/test/java/example/HamcrestAssertionDemo.java]: OK

**Comment by psychsane** [comment]
Fixing build ⭕️ 🐙

**Comment by psychsane** [comment]
@sbrannen I think I have covered most of the 'unrealistic' cases. Please advise on what is left.

**Comment by psychsane** [comment]
@sbrannen quick ping, kindly let me know any more changes that are expected here :)

**Comment by sbrannen** [comment]
> @sbrannen quick ping, kindly let me know any more changes that are expected here :)

Oops. Sorry for the delay. Those updates got lost in my inbox. 

I'll see if I can get back to this PR this week.

**Comment by sbrannen** [comment]
OK. I reviewed your changes, and I think this looks good "as is".

So I'll likely merge this into `master` in the coming days.

Cheers!

**Comment by psychsane** [comment]
> OK. I reviewed your changes, and I think this looks good "as is".
> 
> So I'll likely merge this into `master` in the coming days.
> 
> Cheers!

Super duper! Thank you!!

**Comment by sbrannen** [comment]
> @sbrannen quick ping, kindly let me know any more changes that are expected here :)

Actually, now that I've generated the User Guide, I see that we still have multiple occurrences of `foo`, `bar`, `baz`, etc. in various examples.

Can you please take another pass through the User Guide and replace all such _dummy_ variable names and values with something that appears more realistic?

For example, switching to names of fruits might work our nicely. I did that for an example with `@CsvSource` (search for "apple"). Just a thought... though perhaps you come up with something better than fruits.

Be creative! 😉

**Comment by sbrannen** [comment]
> Ah! Right, I will get rid of the foo-bars. Just need some time :)

I've gone ahead and merged this PR "as is" just to get in into `master`.

Thanks for the contribution! 👍 

If you wish to continue with the "foo-bars", please comment in issue #1651.

**Comment by psychsane** [comment]
> > Ah! Right, I will get rid of the foo-bars. Just need some time :)
> 
> I've gone ahead and merged this PR "as is" just to get it in into `master`.
> 
> Thanks for the contribution! 👍
> 
> If you wish to continue with the "foo-bars", please comment in issue #1651.

Thanks a lot @sbrannen

**Pull request merged**